mapentity/static/mapentity/mapentity.context.js

```diff
@@ -91,9 +93,44 @@ MapEntity.Context = new function() {
 
     };
 
+    self.restoreFilters = function (context, filter) {
+        var $filter = $(filter);
+        $filter.deserialize(context.filter);
+        $filter.find('select').trigger("chosen:updated");
+        $filter.find('select').trigger("change");
+    };
+
+    self.restoreMapContext = function (context, objectsname, kwargs){
+        // This map view change will refresh the list
+        self.restoreMapView(map, context, kwargs);
+
+        // Show layers by their name
+        if (context.maplayers) {
+            var layers = context.maplayers;
+            layers.push(objectsname);
+            $('form.leaflet-control-layers-list input:checkbox').each(function () {
+                if ($.trim($(this).parent().text()) != objectsname) {
+                    $(this).removeAttr('checked');
+                }
+            });
+            for (var i = 0; i < layers.length; i++) {
+                var layer = layers[i];
+                $('form.leaflet-control-layers-list input').each(function () {
+                    if ($.trim($(this).parent().text()) == layer) {
+                        $(this).prop('checked', 'checked');
+                    }
+                });
+            }
+            if ((map.layerscontrol !== undefined) && !!map.layerscontrol._map) {
```

Comment:
unnecessary parentheses
```suggestion
            if (map.layerscontrol !== undefined && !!map.layerscontrol._map) {
```